asciidoc/courses/llm-fundamentals/modules/1-introduction/lessons/1-neo4j-and-genai/lesson.adoc

```diff
@@ -0,0 +1,76 @@
+
+Knowledge graphs typically lean on ontologies, offering structured concepts within a domain and their interrelations. Conversely, while Neo4j can engage with ontologies and offers schema constraints, its standard graphs may not consistently utilize or demand structured semantic frameworks.
+
+With their semantic foundation, knowledge graphs can readily adapt and evolve as they assimilate new information and face domain shifts. In contrast, Neo4j, with its schema-free design, emphasizes flexibility in data accommodation rather than adjustments based on changing knowledge or semantics.
+
+
+== Generative AI & Large Language Models
+
+Generative AI refers to a class of algorithms and models that can generate new content, such as images, text, or even music, based on patterns and examples from existing data. These models learn the underlying structure and distribution of the data and can then generate new samples that resemble the original data.
+
+Generative AI has found applications in a wide range of fields, including image synthesis, text generation, chatbots, and recommendation systems. By leveraging the power of machine learning and deep learning techniques, generative AI models can generate highly realistic and creative outputs.
+
+Large Language Models, most commonly referred to as LLMs, are machine learning models trained on vast amounts of text data to understand and generate human-like text. They can answer questions, create content, and assist with various linguistic tasks by leveraging patterns learned from the data.
+
+
+== Potential Problems
+
+While LLMs provide a lot of potential, they should also be used with caution.
+
+At their core, LLMs are trained to predict the next word(s) in a sequence.
+They convert words into tokens, text into complex vector embeddings; arrays of numbers that can be compared and contrasted.
+In that sense, the model is unaware of what any one word in a sequence means compared to another, only their distance or similarity.
+It is only aware of the similarity.
+
+For this reason, users may find that, under certain conditions, LLMs can produce inaccurate or biased information, may lack context-awareness, and may struggle with tasks requiring structured and relational reasoning.
+This is known as **hallucination**.
+
+Knowledge graphs and graph databases can address these issues by providing LLMs with structured, interconnected data, enabling more accurate context-aware responses and improving reasoning over complex relationships.
+
+
+== Fixing Hallucinations
+
+Knowledge graphs and graph databases can address these issues by providing LLMs with structured, interconnected data, enabling more accurate context-aware responses and improving reasoning over complex relationships.
+
+Providing data to the LLM as part of the instructions passed to the LLM, known as a **prompt**, along with specific instructions, will greatly reduce the LLMs tendency to provide incorrect information.
```

Comment:
Using this approach only the language skills of the LLM are used, not the original training data.